src/xdocs/checks/misc/newlineatendoffile.xml.template

```diff
@@ -99,43 +99,55 @@ StaticMethodCandidateCheck.name = Static Method Candidate
         </p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example1.txt"/>
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example1.java"/>
           <param name="type" value="config"/>
         </macro>
         <p id="Example1-code">Example:</p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example1.txt"/>
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example1.java"/>
+          <param name="type" value="code"/>
+        </macro>
+        <p id="Example2-code">Violation Example:</p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example2.java"/>
           <param name="type" value="code"/>
```

Comment:
examples should go in following way:
...
 <param name="type" value="config"/>
...
 <param name="type" value="code"/>
... 
<param name="type" value="config"/>
...
 <param name="type" value="code"/>

you have 2 "code", please generate website by https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only and make sure that generated content is good and clear for users.

Reply:
I was getting confused by how I should structure the newlineatendoffile.xml.template file and I came up with my solution, let me know if I should make any changes. Here is the website SS after I made the changes:
![final](https://github.com/checkstyle/checkstyle/assets/86093112/7a9e3b7b-7269-4094-ba38-9a7152af3483)


Also after running `mvn clean verify` it was giving me `missing content // xdoc section --` error,  `file does not end with a newline` error & `line too long` or something like that error. I had removed the `// xdoc section --` as you suggested before, should I add them back? And can you help me fix the other errors?